### ISSUE TITLE
Props Name of ReferenceLine

### ISSUE THREAD

**Issue by xile611** [recharts/recharts#35]
Now I need to set three props ( `type`, `value`, `xAxisId` or `yAxisId` ) to describe a reference line. Just like: 
```jsx
  <ReferenceLine type="vertical" value="Page C" xAxisId={1} stroke="green" label="Min PAGE"/>
  <ReferenceLine type="horizontal" value={4000} yAxisId={1}  label="Max" stroke="red" strokeDasharray="3 3"/>
```
I think the props names are a bit obscure. And in order to be consistent , the `value` props of   `ReferenceDot` (Planning Component) would be an object or a string:
```jsx
  <ReferenceDot value={['someVal', 12]}  xAxisId={1}  yAxisId={2} stroke="green" label="Min PAGE"/>
  <ReferenceDot  value={{x: 'someVal', y: 12}} xAxisId={3} yAxisId={1}  label="Max" stroke="red" strokeDasharray="3 3"/>
```

 Maybe we can describe a reference line like this:
```jsx
  <ReferenceLine x="Page C" xAxisId={1} stroke="green" label="Min PAGE"/>
  <ReferenceLine y={4000} yAxisId={1}  label="Max" stroke="red" strokeDasharray="3 3"/>
```
It's just like we describe a horizontal line by `y = 10;`, or a vertical line by `x = 1;` in mathematics.
Then we can describe a reference dot similarly:

```jsx
  <ReferenceDot x="Page C"  y={1000} stroke="green" label="Min PAGE"/>
  <ReferenceLine x="Page C" xAxisId={2}  y={1000} yAxisId={1}  label="Max" stroke="red" strokeDasharray="3 3"/>
```

@arcthur @vanthiyathevan @bkniffler @Cloudo

**Comment by bkniffler** [comment]
I think it is worth considering, it seems more intuitive to, in case of `ReferenceLine`, only define a `x` or `y` prop instead of `type` and `value` or, in case of `ReferenceDot`, define `x` and `y` instead of value pairs. :+1:

**Comment by bkniffler** [comment]
You could still use this for dynamic values, so I really can't think of any problems with that solution right now
```jsx
var values = {x: 'Page C', y: 5};
<ReferenceDot stroke="green" label="Min PAGE" {...values}/>
```

**Comment by arcthur** [comment]
The mathematics formula is great. The `ReferenceLine` has not only two kinds of types, 'vertical' or 'horizontal', but that is most useful. That may be a formula types, such as curve and discontinuous lines. So we can combine formula, and custom render option.

**Comment by xile611** [comment]
Implemented in v0.8.0
